api-gateway/src/main/java/com/example/apigateway/security/AuthServerRoleConverter.java

```diff
@@ -0,0 +1,28 @@
+package com.example.apigateway.security;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AuthServerRoleConverter implements Converter<Jwt, Collection<GrantedAuthority>>{
+
+    @Override
+    public Collection<GrantedAuthority> convert(Jwt jwt) {
+        Map<String, Object> realmAccess = jwt.getClaim("realm_access");
+        if (realmAccess == null || realmAccess.isEmpty()) {
+            return new ArrayList<>();
+        }
+        List<String> roles = (List<String>) realmAccess.get("roles");
+        Collection<GrantedAuthority> grantedAuthorities = roles.stream()
+            .map(roleName -> "ROLE_" + roleName)
+            .map(SimpleGrantedAuthority::new)
+            .collect(Collectors.toList());
```

Comment:
Для информации - с Java 16 можно собирать элементы в список при помощи метода `toList()`.
```suggestion
            .toList();
```